packages/wallet-frontend/src/pages/transfer/pay.tsx

```diff
@@ -14,25 +14,52 @@ import { paySchema, transfersService } from '@/lib/api/transfers'
 import { useDialog } from '@/lib/hooks/useDialog'
 import { SuccessDialog } from '@/components/dialogs/SuccessDialog'
 import { getObjectKeys } from '@/utils/helpers'
-import { useState } from 'react'
 import { paymentPointerService } from '@/lib/api/paymentPointer'
 import { ErrorDialog } from '@/components/dialogs/ErrorDialog'
+import { Controller } from 'react-hook-form'
+import { useState } from 'react'
 
 type PayProps = InferGetServerSidePropsType<typeof getServerSideProps>
 
 export default function Pay({ accounts }: PayProps) {
   const [openDialog, closeDialog] = useDialog()
   const [paymentPointers, setPaymentPointers] = useState<SelectOption[]>([])
   const [balance, setBalance] = useState('')
+
   const payForm = useZodForm({
-    schema: paySchema
+    schema: paySchema,
+    defaultValues: {
+      incomingPaymentUrl: ''
+    }
   })
 
+  const fetchData = async (incomingPaymentUrl: string) => {
+    if (incomingPaymentUrl === '') {
+      return
+    }
+    const tempArray = incomingPaymentUrl.split('/')
+    const incomingPaymentId = tempArray[tempArray.length - 1]
+    const response = await transfersService.getIncomingPaymentDetails(
+      incomingPaymentId
+    )
+    if (response.success && response.data) {
+      const { value, description } = response.data
+      payForm.clearErrors('root')
+      payForm.setValue('amount', value)
+      payForm.setValue('description', description ?? '')
+      payForm.setValue('incomingPaymentUrl', incomingPaymentUrl)
```

Comment:
When we assign a value to the `incomingPaymentUrl` field, is it trimmed automatically? I noticed that the `zod` schema uses the `trim` function, but I'm uncertain if this is reflected visually.